packages/babel-plugin-proposal-async-generator-functions/test/fixtures/nested/generator-in-async/output.js

```diff
@@ -1,22 +1,22 @@
-function f() {
-  return _f.apply(this, arguments);
+function _g() {
+  _g = babelHelpers.wrapAsyncGenerator(function* g() {
+    yield babelHelpers.awaitAsyncGenerator(2);
+    yield 3;
+  });
+  return _g.apply(this, arguments);
 }
 
 function _f() {
-  _f = babelHelpers.asyncToGenerator(function* () {
+  _f = babelHelpers.asyncToGenerator(function* f() {
```

Comment:
Does it work if the asynchronous function refers to itself? This new `f` binding shadows the correct one.